// @skatteetaten/frontend-components/cjs/TopBanner/TopBanner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TopBanner = exports.ExternalHeader = void 0;

var React = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _utils = require("../utils");

var _Icon = require("../Icon");

var _Image = require("../Image");

var _ActionButton = require("../ActionButton");

var _SkeBasis = require("../SkeBasis");

var _i18n = _interopRequireWildcard(require("./../utils/i18n/i18n"));

var _logoSKE = _interopRequireDefault(require("./assets/logoSKE.svg"));

var _logoSKEen = _interopRequireDefault(require("./assets/logoSKEen.svg"));

var _logoLSO = _interopRequireDefault(require("./assets/logoLSO.svg"));

var _logoINK = _interopRequireDefault(require("./assets/logoINK.svg"));

var _logoINKen = _interopRequireDefault(require("./assets/logoINKen.svg"));

var _skeLogoIntern = _interopRequireDefault(require("./assets/ske-logo-intern.svg"));

var _skeLogoInternEn = _interopRequireDefault(require("./assets/ske-logo-intern-en.svg"));

var _External = require("./External.classNames");

var _Internal = require("./Internal.classNames");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

// @ts-ignore TODO
const InternalHeader = props => {
  const styles = (0, _Internal.getClassNames)(props);
  const size = (0, _utils.UseScreen)();
  return /*#__PURE__*/React.createElement("header", {
    className: (0, _classnames.default)(styles.header, props.className),
    id: props.id
  }, /*#__PURE__*/React.createElement("div", {
    className: (0, _classnames.default)(styles.headerLeftContainer, props.slantedAreaClassName)
  }, /*#__PURE__*/React.createElement("a", {
    className: styles.headerLinkContainer,
    href: props.homeUrl,
    onClick: props.onClick
  }, size.gt.md && /*#__PURE__*/React.createElement(_Image.Image, {
    className: styles.headerLogo,
    src: props.language === 'en' ? _skeLogoInternEn.default : _skeLogoIntern.default,
    alt: "Skatteetaten logo"
  }), /*#__PURE__*/React.createElement(_Icon.Icon, {
    className: styles.headerIcon,
    iconName: props.icon
  }), /*#__PURE__*/React.createElement("div", {
    className: styles.headerLink
  }, props.homeText))), /*#__PURE__*/React.createElement("div", {
    className: styles.headerDiagonal
  }), /*#__PURE__*/React.createElement("div", {
    className: styles.headerRightContainer
  }, typeof props.title === 'string' ? /*#__PURE__*/React.createElement("h1", null, props.title) : /*#__PURE__*/React.createElement("div", {
    className: styles.elementTitle
  }, props.title), /*#__PURE__*/React.createElement("div", null, props.children)));
}; // @ts-ignore


const ExternalHeaderContent = _ref => {
  let {
    styles
  } = _ref,
      props = _objectWithoutProperties(_ref, ["styles"]);

  if (props.children) {
    return props.children;
  }

  if (props.compact) {
    return /*#__PURE__*/React.createElement("h1", {
      className: styles.title
    }, props.title);
  }

  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(_ActionButton.ActionButton, {
    href: props.homeUrl,
    onClick: props.onClick,
    className: styles.linkButton,
    icon: "ArrowBack",
    role: "link"
  }, props.homeText), /*#__PURE__*/React.createElement("h1", {
    className: styles.title
  }, props.title));
};
/*
 * visibleName TopBanner (Topp)
 */


const TopBanner = props => {
  const {
    external
  } = props,
        rest = _objectWithoutProperties(props, ["external"]);

  return external ? /*#__PURE__*/React.createElement(ExternalHeader, rest) : /*#__PURE__*/React.createElement(InternalHeader, rest);
};

exports.TopBanner = TopBanner;

const ExternalHeader = props => {
  const styles = (0, _External.getClassNames)(props, 'SKE'); // @ts-ignore

  const {
    logo,
    headerMain,
    contentWrapper
  } = styles;
  const compactHeight = props.compact ? 55 : 68;
  const isLink = props.logoLink;

  if (props.language) {
    _i18n.default.changeLanguage(props.language);
  }

  const logoImageElement = (brand, showAltText = true) => {
    switch (brand) {
      case 'SKE':
        return /*#__PURE__*/React.createElement(_Image.Image, {
          src: props.language === 'en' ? _logoSKEen.default : _logoSKE.default,
          height: compactHeight,
          alt: isLink ? (0, _i18n.t)('topbanner.ske.logoLink') : (0, _i18n.t)('topbanner.ske.logo')
        });

      case 'INK':
        return /*#__PURE__*/React.createElement(_Image.Image, {
          src: props.language === 'en' ? _logoINKen.default : _logoINK.default,
          height: compactHeight,
          alt: isLink ? (0, _i18n.t)('topbanner.ink.logoLink') : (0, _i18n.t)('topbanner.ink.logo')
        });

      case 'LSO':
        return /*#__PURE__*/React.createElement(_Image.Image, {
          src: _logoLSO.default,
          height: compactHeight,
          alt: isLink ? (0, _i18n.t)('topbanner.lso.logoLink') : (0, _i18n.t)('topbanner.lso.logo')
        });

      default:
        return {};
    }
  };

  return /*#__PURE__*/React.createElement(_SkeBasis.BrandContext.Consumer, null, ({
    tag
  }) => /*#__PURE__*/React.createElement("header", {
    className: (0, _classnames.default)((0, _External.getClassNames)(props, tag).header, props.className),
    id: props.id
  }, props.topStripe, /*#__PURE__*/React.createElement("div", {
    className: headerMain
  }, /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("div", {
    className: logo
  }, props.logoLink ? /*#__PURE__*/React.createElement("a", {
    href: props.logoLinkUrl
  }, logoImageElement(tag, true)) : logoImageElement(tag, true))), /*#__PURE__*/React.createElement("div", {
    className: contentWrapper
  }, /*#__PURE__*/React.createElement(ExternalHeaderContent, _extends({
    styles: styles
  }, props))))));
};

exports.ExternalHeader = ExternalHeader;
TopBanner.defaultProps = {
  title: undefined,
  homeUrl: undefined,
  homeText: undefined,
  slantedAreaWidth: undefined,
  icon: 'Home',
  external: false,
  compact: false,
  logoLink: true,
  logoLinkUrl: 'https://www.skatteetaten.no',
  language: undefined
};
