// @skatteetaten/frontend-components/cjs/Table/SumRow/SumRow.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _spacing = _interopRequireDefault(require("../../utils/designtokens/_spacing.json"));

var _fontSizes = _interopRequireDefault(require("../../utils/designtokens/_fontSizes.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const getClassNames = compactTable => {
  return (0, _mergeStyles.mergeStyleSets)({
    sumRow: {
      verticalAlign: 'middle'
    },
    sumCell: {
      fontSize: compactTable ? _fontSizes.default['ske-font-size-s'] : 'inherit',
      textAlign: 'end',
      verticalAlign: 'inherit',
      padding: compactTable ? `${_spacing.default['ske-spacing-sm']} ${_spacing.default['ske-spacing-sm']}` : `${_spacing.default['ske-spacing-lg']} ${_spacing.default['ske-spacing-lg']}`,
      selectors: {
        '&.sumCellIsText': {
          fontWeight: _fontSizes.default['ske-font-weight-bold']
        },
        '&.sumCellIsSum': {
          fontWeight: _fontSizes.default['ske-font-weight-regular']
        }
      }
    }
  });
};

exports.getClassNames = getClassNames;
