// @skatteetaten/frontend-components/cjs/TopStripe/LanguagePicker/LanguagePicker.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _fontSizes = _interopRequireDefault(require("../../utils/designtokens/_fontSizes.json"));

var _breakpoints = _interopRequireDefault(require("../../utils/designtokens/_breakpoints.json"));

var _spacing = _interopRequireDefault(require("../../utils/designtokens/_spacing.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const getClassNames = () => {
  return (0, _mergeStyles.mergeStyleSets)({
    languageButton: {
      paddingLeft: '2.25rem !important',
      width: '100%',
      selectors: {
        [`@media (min-width: ${_breakpoints.default['ske-breakpoint-sm']})`]: {
          paddingLeft: `${_spacing.default['ske-spacing-xl']} !important`
        }
      }
    },
    languageButtonContent: {
      fontSize: _fontSizes.default['ske-font-size-l'],
      selectors: {
        [`@media (min-width: ${_breakpoints.default['ske-breakpoint-sm']})`]: {
          fontSize: `${_fontSizes.default['ske-font-size-s']}`
        }
      }
    },
    languageButtonIsNotSelected: {
      fontWeight: ` ${_fontSizes.default['ske-font-weight-bold']}`,
      selectors: {
        [`@media (min-width: ${_breakpoints.default['ske-breakpoint-sm']})`]: {
          fontWeight: `${_fontSizes.default['ske-font-weight-regular']}`
        }
      }
    },
    checkIcon: {
      left: '0',
      width: '0.875rem',
      margin: `0 ${_spacing.default['ske-spacing-md']}`,
      fontSize: _fontSizes.default['ske-font-size-xl'],
      display: 'inline',
      position: 'absolute',
      selectors: {
        [`@media (min-width: ${_breakpoints.default['ske-breakpoint-sm']})`]: {
          fontSize: _fontSizes.default['ske-font-size-m'],
          padding: `0 ${_spacing.default['ske-spacing-sm']}`,
          left: 'auto',
          width: 'auto',
          marginLeft: '-1.5rem'
        }
      }
    },
    languageButtonFlag: {
      marginRight: _spacing.default['ske-spacing-sm'],
      verticalAlign: 'middle'
    },
    languageButtonText: {
      marginLeft: _spacing.default['ske-spacing-sm']
    }
  });
};

exports.getClassNames = getClassNames;
