// @skatteetaten/frontend-components/FooterContent/FooterContent.js
function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import * as React from 'react';
import { Image } from '../Image';
import { getClassNames, getLogoClassNames } from './FooterContent.classNames';
import footerDekor from './footerDekor';
import logo from './assets/ske-logo.svg';
import logoEn from './assets/ske-logo-en.svg';
import logoInk from './assets/ink-logo.svg';
import { BrandContext } from '../SkeBasis';

const Logo = () => {
  return /*#__PURE__*/React.createElement(Image, {
    src: logo,
    className: getLogoClassNames(),
    height: "74px",
    alt: "Skatteetaten logo"
  });
};

const LogoEn = () => {
  return /*#__PURE__*/React.createElement(Image, {
    src: logoEn,
    className: getLogoClassNames(),
    height: "74px",
    alt: "Norwegian Tax Authorities logo"
  });
};

const LogoInk = () => {
  return /*#__PURE__*/React.createElement(Image, {
    src: logoInk,
    className: getLogoClassNames(),
    height: "63px",
    alt: "Statens innkreving - en del av Skatteetaten, logo"
  });
};
/*
 * visibleName FooterContent (Bunn)
 */


export class FooterContent extends React.PureComponent {
  /** Engelsk logo */

  /** Norsk Logo */

  /** Statens innkreving logo */
  render() {
    const {
      children,
      className,
      ariaLabel
    } = this.props;
    const styles = getClassNames();
    return /*#__PURE__*/React.createElement(BrandContext.Consumer, null, ({
      tag,
      primaryColor,
      secondaryColor
    }) => /*#__PURE__*/React.createElement("div", {
      className: className,
      "aria-label": ariaLabel
    }, /*#__PURE__*/React.createElement("div", {
      className: styles.footerDecorContainer,
      "aria-hidden": "true"
    }, footerDekor(primaryColor, secondaryColor)), /*#__PURE__*/React.createElement("footer", {
      className: styles['footerWrapper' + tag]
    }, /*#__PURE__*/React.createElement("div", {
      className: styles['footerContent' + tag]
    }, children))));
  }

}

_defineProperty(FooterContent, "LogoEn", LogoEn);

_defineProperty(FooterContent, "Logo", Logo);

_defineProperty(FooterContent, "LogoInk", LogoInk);
