// @skatteetaten/frontend-components/cjs/CheckBox/CheckBox.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _react = require("@fluentui/react");

var _utils = require("../utils");

var _colors = _interopRequireDefault(require("../utils/designtokens/_colors.json"));

var _fontSizes = _interopRequireDefault(require("../utils/designtokens/_fontSizes.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const getClassNames = function getClassNames(error) {
  const theme = (0, _react.getTheme)();
  const inset = -4;
  const radius = '0';
  return (0, _mergeStyles.mergeStyles)([(0, _utils.getFocusStyle)(theme, inset, 'relative', radius), {
    displayName: 'SkeAvkrysningsboks',
    selectors: {
      '&.ms-Checkbox': {
        padding: '4px',
        border: error === true ? `2px solid ${_colors.default['ske-color-status-error']}` : 'none'
      },
      '&.is-enabled .ms-Checkbox-checkbox': {
        borderRadius: '0px',
        content: '',
        display: 'inline-block',
        border: `2px solid ${_colors.default['ske-color-black-100']}`,
        cursor: 'pointer',
        fontSize: _fontSizes.default['ske-font-size-m'],
        backgroundColor: _colors.default['ske-color-white-100']
      },
      'span.ms-Checkbox-text': {
        fontSize: _fontSizes.default['ske-font-size-m']
      },
      '&.is-checked .ms-Checkbox-checkbox': {
        border: 'none',
        fontSize: _fontSizes.default['ske-font-size-m'],
        backgroundColor: _colors.default['ske-color-interactive']
      },
      '&.is-disabled .ms-Checkbox-checkbox': {
        borderRadius: 0,
        paddingTop: 0,
        fontSize: _fontSizes.default['ske-font-size-m'],
        backgroundColor: _colors.default['ske-color-grey-10'],
        border: `2px solid ${_colors.default['ske-color-grey-50']}`
      },
      '&.is-disabled i.ms-Checkbox-checkmark': {
        paddingTop: 0,
        color: _colors.default['ske-color-grey-50'],
        paddingLeft: 0
      },
      'i.ms-Checkbox-checkmark': {
        paddingLeft: '1.5px',
        paddingTop: '1.5px',
        fontSize: _fontSizes.default['ske-font-size-m']
      },
      '&:hover .ms-Checkbox-checkmark': {
        opacity: 0
      },
      '&.is-checked:hover .ms-Checkbox-checkmark': {
        opacity: 1
      },
      '& i': {
        fontWeight: _fontSizes.default['ske-font-weight-bold']
      },
      '.ms-Checkbox-label:hover i': {
        opacity: 0
      }
    }
  }]);
};

exports.getClassNames = getClassNames;
