// @skatteetaten/frontend-components/cjs/TopStripe/TopStripe.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TopStripeContext = exports.TopStripe = void 0;

var React = _interopRequireWildcard(require("react"));

var _TopStripe = require("./TopStripe.classNames");

var _classnames = _interopRequireDefault(require("classnames"));

var _TopStripeMenu = require("./TopStripeMenu");

var _TopStripeButton = require("./TopStripeButton");

var _TopStripeLink = require("./TopStripeLink");

var _LanguagePicker = require("./LanguagePicker");

var _TopStripeUser = require("./TopStripeUser");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

const TopStripeContext = /*#__PURE__*/React.createContext({
  open: -1
});
/*
 * visibleName TopStripe (Toppstripe)
 */

exports.TopStripeContext = TopStripeContext;

const TopStripe = props => {
  const topStripeElements = [_TopStripeMenu.TopStripeMenu, _TopStripeButton.TopStripeButton, _TopStripeLink.TopStripeLink, _TopStripeUser.TopStripeUser, _LanguagePicker.LanguagePicker];
  const notOpenIndex = -1;
  const topRef = /*#__PURE__*/React.createRef();
  const [openIndex, setOpenIndex] = React.useState(notOpenIndex);

  const setOpen = index => {
    if (openIndex === index) {
      setOpenIndex(notOpenIndex);
    } else {
      setOpenIndex(index);
    }
  };

  const {
    children,
    className,
    contentWidth
  } = props,
        rest = _objectWithoutProperties(props, ["children", "className", "contentWidth"]);

  const styles = (0, _TopStripe.getClassNames)(contentWidth);
  const showOverlay = openIndex !== notOpenIndex ? styles.overlayShow : '';

  const handleClickOutside = e => {
    const eventPaths = e.composedPath ? e.composedPath() : [];
    const target = eventPaths.length > 0 ? eventPaths[0] : e.target;
    const node = topRef.current;

    if (node && node.contains(target)) {
      // inside click
      return;
    } // outside click


    setOpen(notOpenIndex);
  };

  const handleEscape = e => {
    // Match escape key
    if (e.keyCode === 27) {
      setOpen(notOpenIndex);
    }
  };

  React.useEffect(() => {
    document.addEventListener('mousedown', handleClickOutside);
    document.addEventListener('keydown', handleEscape);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
      document.removeEventListener('keydown', handleEscape);
    };
  });
  return /*#__PURE__*/React.createElement("div", {
    className: styles.topStripe
  }, /*#__PURE__*/React.createElement("div", {
    className: (0, _classnames.default)(styles.overlay, showOverlay)
  }), /*#__PURE__*/React.createElement("div", {
    className: styles.background
  }, /*#__PURE__*/React.createElement(TopStripeContext.Provider, {
    value: {
      open: openIndex,
      setOpen: setOpen,
      closeMenu: () => setOpenIndex(notOpenIndex)
    }
  }, /*#__PURE__*/React.createElement("ul", _extends({
    ref: topRef,
    className: (0, _classnames.default)(styles.topStripeContainer, className)
  }, rest), React.Children.map(children, (child, index) => child ? /*#__PURE__*/React.createElement("li", {
    className: (0, _classnames.default)(styles.topStripeElement, {
      [styles.loggedInUser]: child.type === _TopStripeUser.TopStripeUser,
      [styles.hideOnMobile]: topStripeElements.includes(child.type) && !child.props.showOnMobile
    })
  }, /*#__PURE__*/React.cloneElement(child, {
    index
  })) : null)))));
};

exports.TopStripe = TopStripe;
