// @skatteetaten/frontend-components/cjs/Chip/Chip.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _react = require("@fluentui/react");

var _mergeStyles = require("@fluentui/merge-styles");

const getClassNames = function getClassNames(props) {
  const palette = (0, _react.getTheme)().palette;
  return (0, _mergeStyles.mergeStyles)({
    displayName: `ske-chip`,
    display: 'inline-block',
    color: palette.skeColor.blackAlt,
    fontWeight: '400',
    border: `1px solid ${palette.skeColor.white}`,
    verticalAlign: 'middle',
    padding: '3px 12px',
    fontSize: props.size,
    borderRadius: '5px',
    backgroundColor: palette.skeColor[props.type],
    selectors: {
      '& > div': {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center'
      },
      '& a': {
        textDecoration: 'underlined',
        color: palette.skeColor.blackAlt
      },
      '& i': {
        verticalAlign: 'text-bottom'
      }
    }
  });
};

exports.getClassNames = getClassNames;
