// @skatteetaten/frontend-components/cjs/Pagination/Pagination.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _react = require("@fluentui/react");

var _mergeStyles = require("@fluentui/merge-styles");

var _utils = require("../utils");

const getClassNames = () => {
  const palette = (0, _react.getTheme)().palette;
  return (0, _mergeStyles.mergeStyleSets)({
    activePage: {
      color: `${palette.skeColor.darkGrey} !important`,
      borderBottom: 'none !important',
      paddingTop: '3px !important',
      paddingBottom: '6px !important'
    },
    linkIcons: {
      fontSize: _utils.FontSizes.xLarge,
      verticalAlign: 'middle'
    },
    pageNumber: {
      borderBottom: `2px solid ${palette.skeColor.lightBlue} !important`,
      paddingTop: '3px !important'
    },
    paginationContainer: {
      paddingTop: '24px',
      fontSize: _utils.FontSizes.medium,
      selectors: {
        button: {
          border: 'none',
          background: 'transparent',
          fontSize: _utils.FontSizes.medium
        },
        nav: {
          float: 'right',
          selectors: {
            '@media only screen and (max-width: 479px)': {
              margin: '0 auto',
              float: 'none'
            }
          }
        },
        p: {
          marginTop: '7px',
          float: 'left',
          selectors: {
            '@media only screen and (max-width: 479px)': {
              margin: '0 auto',
              float: 'none'
            }
          }
        },
        ul: {
          listStyle: 'none',
          display: 'inline-flex',
          paddingLeft: '0',
          margin: 0
        },
        li: {
          display: 'inline',
          selectors: {
            button: {
              color: palette.skeColor.blue,
              fontWeight: 'bold',
              padding: '3px 1px 3px 1px',
              margin: '4px 6px ',
              cursor: 'pointer',
              verticalAlign: 'middle'
            },
            'button:focus': {
              color: palette.skeColor.darkBlue,
              backgroundColor: palette.skeColor.lightBlue,
              outline: 'none'
            },
            'button:hover': {
              color: palette.skeColor.darkBlue,
              backgroundColor: palette.skeColor.lightBlue
            }
          }
        },
        '@media only screen and (max-width: 479px)': {
          display: 'grid'
        }
      }
    }
  });
};

exports.getClassNames = getClassNames;
