// @skatteetaten/frontend-components/cjs/TopStripe/TopStripeMenu/TopStripeMenu.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _colors = _interopRequireDefault(require("../../utils/designtokens/_colors.json"));

var _fontSizes = _interopRequireDefault(require("../../utils/designtokens/_fontSizes.json"));

var _spacing = _interopRequireDefault(require("../../utils/designtokens/_spacing.json"));

var _breakpoints = _interopRequireDefault(require("../../utils/designtokens/_breakpoints.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const getClassNames = () => {
  return (0, _mergeStyles.mergeStyleSets)({
    topStripeMenuIcon: {
      fontSize: _fontSizes.default['ske-font-size-m'],
      verticalAlign: 'middle',
      position: 'absolute',
      bottom: '0.625rem',
      left: '0.25rem'
    },
    topStripeMenuShowChevron: {
      paddingRight: '1.375rem !important'
    },
    topStripeMenuHasIcon: {
      paddingLeft: '1.375rem !important'
    },
    topStripeMenuTitle: {
      textDecoration: 'underline',
      display: 'inline'
    },
    topStripeMenuChevronIcon: {
      position: 'absolute',
      fontSize: _fontSizes.default['ske-font-size-xl'],
      bottom: '0.5rem'
    },
    topStripeMenuDropdownContainer: {
      zIndex: '200',
      position: 'absolute',
      top: '2.625rem',
      marginLeft: _spacing.default['ske-spacing-md'],
      backgroundColor: _colors.default['ske-color-black-100'],
      minWidth: '100vw',
      padding: 0,
      paddingInlineStart: 0,
      paddingInlineEnd: 0,
      margin: 0,
      selectors: {
        'a, i': {
          borderWidth: 0
        },
        [`@media (min-width: ${_breakpoints.default['ske-breakpoint-sm']}) and (max-width: 1023px)`]: {
          left: 'auto',
          right: 0
        },
        [`@media (min-width: ${_breakpoints.default['ske-breakpoint-sm']})`]: {
          top: '2.375rem',
          padding: `${_spacing.default['ske-spacing-md']} ${_spacing.default['ske-spacing-lg']}`,
          minWidth: 'auto'
        }
      }
    },
    topStripeMenuDropdownElementContainer: {
      position: 'relative'
    },
    topStripeMenuDropdownElement: {
      fontSize: _fontSizes.default['ske-font-size-s'],
      background: 'none',
      border: 'none',
      color: 'inherit',
      textDecoration: 'none !important',
      transition: 'none'
    },
    topStripeMenuDropdownElementIcon: {
      position: 'absolute',
      top: 14,
      left: 4
    },
    topStripeMenuDropdownCloseButton: {
      width: '100%',
      display: 'block',
      margin: '0 auto',
      selectors: {
        '.ms-Button-flexContainer': {
          justifyContent: 'center'
        },
        ':hover, :focus, :hover i, :focus i': {
          backgroundColor: `${_colors.default['ske-color-white-100']} !important`,
          color: `${_colors.default['ske-color-black-100']} !important`,
          outline: 'none'
        },
        i: {
          color: `${_colors.default['ske-color-white-100']} !important`
        }
      }
    }
  });
};

exports.getClassNames = getClassNames;
