// @skatteetaten/frontend-components/cjs/TopStripe/TopStripeButton/TopStripeButton.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _fontSizes = _interopRequireDefault(require("../../utils/designtokens/_fontSizes.json"));

var _colors = _interopRequireDefault(require("../../utils/designtokens/_colors.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const getClassNames = () => {
  return (0, _mergeStyles.mergeStyleSets)({
    topStripeButton: {
      fontFamily: 'inherit',
      fontSize: _fontSizes.default['ske-font-size-s'],
      background: 'inherit',
      border: 'none',
      color: _colors.default['ske-color-white-100'],
      transition: 'none',
      textDecoration: 'underline',
      textAlign: 'left',
      padding: '0.75rem 0.5rem',
      margin: 0,
      whiteSpace: 'nowrap',
      selectors: {
        ':hover, :focus': {
          backgroundColor: _colors.default['ske-color-white-100'],
          color: _colors.default['ske-color-black-100'],
          transition: 'none',
          outline: 'none'
        }
      }
    }
  });
};

exports.getClassNames = getClassNames;
