// @skatteetaten/frontend-components/Dropdown/Dropdown.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import classnames from 'classnames';
import { Dropdown as FabricDropdown, DropdownMenuItemType } from '@fluentui/react';
import * as React from 'react';
import { getClassNames, getCalloutStyles } from './Dropdown.classNames';
import { generateId } from '../utils';
import { Icon } from '../Icon';
import { LabelWithCallout } from '../LabelWithCallout';

/*
 * visibleName Dropdown (Nedtrekksliste)
 */
export const Dropdown = props => {
  const {
    children,
    errorMessage,
    label,
    labelButtonAriaLabel,
    help,
    onRenderLabel,
    required = false,
    className,
    id,
    labelWithCalloutProps,
    onCalloutToggle,
    calloutProps,
    readOnly,
    requiredWithMark = false
  } = props,
        rest = _objectWithoutProperties(props, ["children", "errorMessage", "label", "labelButtonAriaLabel", "help", "onRenderLabel", "required", "className", "id", "labelWithCalloutProps", "onCalloutToggle", "calloutProps", "readOnly", "requiredWithMark"]);

  const generatedId = generateId();
  const mainId = id ? id : 'dropdown-' + generatedId;
  const inputId = mainId + '-input';
  const labelId = mainId + '-label';
  const styles = getClassNames(props);
  const dropdownStyles = getCalloutStyles();
  return /*#__PURE__*/React.createElement("div", {
    id: mainId
  }, /*#__PURE__*/React.createElement(LabelWithCallout, {
    id: labelId,
    inputId: readOnly ? inputId : inputId + '-option',
    label: label,
    requiredMark: requiredWithMark,
    buttonAriaLabel: labelButtonAriaLabel,
    help: help,
    onCalloutToggle: onCalloutToggle,
    calloutProps: _objectSpread({}, labelWithCalloutProps?.calloutProps)
  }), readOnly ? /*#__PURE__*/React.createElement("input", {
    id: inputId,
    type: "text",
    readOnly: true,
    className: styles.readOnly,
    "aria-invalid": errorMessage ? true : false,
    value: props.options.filter(option => option.key === (props.selectedKey || props.defaultSelectedKey))[0].text
  }) : /*#__PURE__*/React.createElement(FabricDropdown, _extends({}, rest, {
    ariaLabel: label,
    "aria-invalid": errorMessage ? true : false,
    id: inputId,
    required: required || requiredWithMark,
    className: classnames(styles.main, className),
    styles: dropdownStyles,
    onRenderCaretDown: () => /*#__PURE__*/React.createElement(Icon, {
      iconName: 'ChevronDown'
    }),
    errorMessage: errorMessage,
    calloutProps: _objectSpread({}, calloutProps)
  }), children));
}; // @ts-ignore

Dropdown.ItemType = DropdownMenuItemType;
