// @skatteetaten/frontend-components/cjs/ButtonLink/ButtonLink.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _react = require("@fluentui/react");

var _utils = require("../utils");

const getClassNames = () => {
  const palette = (0, _react.getTheme)().palette;
  return (0, _mergeStyles.mergeStyleSets)({
    buttonLink: {
      display: 'inline-block',
      borderColor: palette.skeColor.blue,
      background: palette.skeColor.blue,
      color: palette.skeColor.white,
      boxShadow: `0 8px 6px -6px ${palette.skeColor.grey30}`,
      fontSize: _utils.FontSizes.largePlus,
      borderWidth: '3px',
      borderRadius: '7px',
      borderStyle: 'solid',
      fontWeight: _utils.FontWeights.semibold,
      padding: '25px',
      maxWidth: 'calc((75*550px)/100)',
      textDecoration: 'none',
      transition: '0.2s',
      selectors: {
        '&:hover': {
          borderColor: palette.skeColor.interactiveDark,
          background: palette.skeColor.interactiveDark,
          color: palette.white
        },
        '&:focus': {
          outline: '2px solid ' + palette.skeColor.interactive,
          outlineOffset: '3px'
        },
        '&:active': {
          transform: 'translateY(2px)',
          transitionTimingFunction: 'ease',
          transition: '0.15s',
          boxShadow: 'none'
        }
      }
    }
  });
};

exports.getClassNames = getClassNames;
