// @skatteetaten/frontend-components/cjs/ErrorMessage/ErrorMessage.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _react = require("@fluentui/react");

var _utils = require("../utils");

const errorIcon = "'" + _utils.SkeIcons.icons.Error + "'";

const getClassNames = function getClassNames() {
  const theme = (0, _react.getTheme)();
  const {
    fonts
  } = theme;
  const palette = theme.palette;
  const classNames = (0, _react.getGlobalClassNames)({}, theme);
  return (0, _mergeStyles.mergeStyles)([// @ts-ignore TODO
  classNames.error, _react.AnimationClassNames.slideDownIn20, fonts.small, {
    display: 'block',
    color: palette.skeColor.error,
    fontWeight: 500,
    lineHeight: '1rem',
    paddingTop: 5,
    selectors: {
      ':before': {
        fontFamily: _utils.SkeIcons.fontFace.fontFamily,
        fontSize: fonts.small,
        content: errorIcon,
        marginRight: 3,
        verticalAlign: 'text-bottom'
      }
    }
  }]);
};

exports.getClassNames = getClassNames;
