// @skatteetaten/frontend-components/NavigationTile/NavigationTile.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import classnames from 'classnames';
import * as React from 'react';
import { NavigationContent } from './NavigationContent';
import { getClassNames } from './NavigationTile.classNames';

/*
 * visibleName NavigationTile (Forsideknapp)
 */
export const NavigationTile = props => {
  const {
    children,
    contents,
    className,
    type,
    alignIcon,
    alignTitle,
    alignDescription,
    ariaLabel,
    headingLevel,
    useButtons
  } = props,
        rest = _objectWithoutProperties(props, ["children", "contents", "className", "type", "alignIcon", "alignTitle", "alignDescription", "ariaLabel", "headingLevel", "useButtons"]);

  const styles = getClassNames(props);
  return /*#__PURE__*/React.createElement("nav", _extends({
    "aria-label": ariaLabel
  }, rest, {
    className: classnames(styles.nav, getClassNames(props), className)
  }), /*#__PURE__*/React.createElement("ul", null, contents && contents.map((_ref, index) => {
    let rest = _extends({}, _ref);

    return /*#__PURE__*/React.createElement(NavigationContent, _extends({
      key: index + '.' + rest.to,
      className: styles.content,
      headingLevel: headingLevel,
      useButtons: useButtons
    }, rest));
  }), React.Children.map(children, (child, index) => {
    if ( /*#__PURE__*/React.isValidElement(child)) {
      return /*#__PURE__*/React.createElement(NavigationContent, _extends({
        key: index + '.' + child.props.to,
        className: styles.content,
        headingLevel: headingLevel,
        useButtons: useButtons
      }, child.props), child.props.children);
    }
  })));
};
NavigationTile.defaultProps = {
  alignDescription: 'center',
  alignIcon: 'center',
  alignTitle: 'center',
  type: 'center'
};
