// @skatteetaten/frontend-components/cjs/StepList/Step/Step.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Step = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _utils = require("../../utils");

var _ActionButton = require("../../ActionButton");

var _Grid = require("../../Grid");

var _Icon = require("../../Icon");

var _StepList = require("../StepList.classNames");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

const NumberIcon = props => {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: props.styles.numberWrapper
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: props.styles.stepNumber
  }, /*#__PURE__*/_react.default.createElement("span", null, props.stepType === 'result' ? /*#__PURE__*/_react.default.createElement(_Icon.Icon, {
    iconName: props.resultIcon,
    style: {
      fontSize: '24px'
    }
  }) : '')));
};
/*
 * visibleName Step (Enkeltsteg)
 */


const Step = props => {
  const {
    stepTitle,
    stepNumber,
    children,
    stepId,
    resultIcon,
    className,
    stepType,
    actionBtn,
    gridSpacing,
    headingLevel = 2
  } = props;

  const [styles, setStyles] = _react.default.useState((0, _StepList.getClassNames)(props));

  const size = (0, _utils.UseScreen)();
  const headingId = stepId ? stepId + '-heading' : undefined;

  _react.default.useEffect(() => {
    setStyles((0, _StepList.getClassNames)(_objectSpread(_objectSpread({}, props), {}, {
      showStep: typeof props.showStep === 'boolean' ? props.showStep : true,
      stepType: props.stepType || 'passive'
    })));
  }, [props]);

  return /*#__PURE__*/_react.default.createElement("div", {
    key: stepNumber,
    className: (0, _classnames.default)(styles.wrapperStep, className)
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid.Row, {
    rowSpacing: _Grid.Grid.SPACE_NONE
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid.Col, {
    noSpacing: true
  }, /*#__PURE__*/_react.default.createElement(_Grid.Grid.Row, {
    rowSpacing: _Grid.Grid.SPACE_NONE
  }, gridSpacing && /*#__PURE__*/_react.default.createElement(_Grid.Grid.Col, {
    noSpacing: true,
    sm: 0,
    lg: 1,
    xxxl: 2
  }), /*#__PURE__*/_react.default.createElement(_Grid.Grid.Col, {
    sm: 12,
    lg: gridSpacing ? 10 : undefined,
    xxxl: gridSpacing ? 8 : undefined
  }, stepNumber && stepNumber > 1 && /*#__PURE__*/_react.default.createElement("span", {
    className: (0, _classnames.default)(styles.stepLineTop)
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: (0, _classnames.default)(styles.stepLine)
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: (0, _classnames.default)({
      [styles.arrowLine]: stepType === 'next'
    })
  }), stepType !== 'next' && /*#__PURE__*/_react.default.createElement(NumberIcon, {
    styles: styles,
    resultIcon: resultIcon,
    stepType: stepType
  }), stepType !== 'result' && /*#__PURE__*/_react.default.createElement("hr", {
    className: (0, _classnames.default)(styles.divider)
  }), /*#__PURE__*/_react.default.createElement("div", {
    id: stepId
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: styles.stepContent
  }, stepTitle && /*#__PURE__*/_react.default.createElement(_utils.Heading, {
    id: headingId,
    text: stepTitle,
    level: headingLevel,
    tabIndex: stepId ? -1 : undefined,
    className: styles.title
  }, !size.gt.sm && stepType !== 'next' && /*#__PURE__*/_react.default.createElement(NumberIcon, {
    styles: styles,
    resultIcon: resultIcon,
    stepType: stepType
  }), /*#__PURE__*/_react.default.createElement("span", {
    className: (0, _classnames.default)(styles.titleText)
  }, stepTitle)), /*#__PURE__*/_react.default.createElement("div", {
    className: styles.stepContentInner
  }, children, actionBtn && /*#__PURE__*/_react.default.createElement(_ActionButton.ActionButton, {
    icon: actionBtn.icon,
    className: styles.stepAction,
    onClick: actionBtn.event,
    ariaLabel: actionBtn.ariaLabel
  }, actionBtn.text))))), gridSpacing && /*#__PURE__*/_react.default.createElement(_Grid.Grid.Col, {
    noSpacing: true,
    sm: 0,
    lg: 1,
    xxxl: 2
  })))));
};

exports.Step = Step;
