// @skatteetaten/frontend-components/cjs/Layout/Nav.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Nav = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Grid = require("../Grid");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class Nav extends _Grid.Grid.Col {
  render() {
    return super.render();
  }

}

exports.Nav = Nav;

_defineProperty(Nav, "displayName", 'Nav');

_defineProperty(Nav, "propTypes", {
  tag: _propTypes.default.string,
  sm: _propTypes.default.number,
  lg: _propTypes.default.number,
  xl: _propTypes.default.number,
  xxl: _propTypes.default.number,
  xxxl: _propTypes.default.number,
  smPush: _propTypes.default.number,
  lgPush: _propTypes.default.number,
  xlPush: _propTypes.default.number,
  xxlPush: _propTypes.default.number,
  xxxlPush: _propTypes.default.number,
  smPull: _propTypes.default.number,
  lgPull: _propTypes.default.number,
  xlPull: _propTypes.default.number,
  xxlPull: _propTypes.default.number,
  xxxlPull: _propTypes.default.number
});

_defineProperty(Nav, "defaultProps", {
  tag: 'nav',
  lg: 2,
  xl: 2 //lgPull: 10,
  //xlPull: 8

});
