// @skatteetaten/frontend-components/Button/Button.classNames.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { mergeStyles } from '@fluentui/merge-styles';
import { getTheme } from '@fluentui/react';
import { getFocusStyle } from '../utils';
import designtokenColors from '../utils/designtokens/_colors.json';
import designtokenFontSizes from '../utils/designtokens/_fontSizes.json';
import designtokenBreakpoints from '../utils/designtokens/_breakpoints.json';

function getTypeColor(props) {
  const radius = '200px';
  const interactiveColor = designtokenColors['ske-color-interactive'];
  const whiteColor = designtokenColors['ske-color-white-100'];
  const sizeNormal = {
    height: 'auto',
    minHeight: '32px',
    padding: '7px 15px'
  };

  switch (props.buttonStyle) {
    case 'primaryCornered':
      return _objectSpread(_objectSpread({
        borderRadius: '7px',
        borderColor: interactiveColor,
        background: interactiveColor,
        color: whiteColor
      }, sizeNormal), {}, {
        selectors: {
          [`@media  only screen and (max-width: ${designtokenBreakpoints['ske-breakpoint-md']})`]: {
            width: props.mobileFullWidth ? '100%' : 'auto'
          }
        }
      });

    case 'callToAction':
      return {
        borderColor: interactiveColor,
        background: interactiveColor,
        color: whiteColor,
        boxShadow: `0 8px 6px -6px ${designtokenColors['ske-color-grey-30']}`,
        fontSize: designtokenFontSizes['ske-font-size-xl'],
        borderWidth: '3px',
        borderRadius: '8px',
        fontWeight: designtokenFontSizes['ske-font-weight-semibold'],
        padding: '25px',
        height: '80px',
        maxWidth: 'calc((75*550px)/100)'
      };

    case 'primary':
      return _objectSpread(_objectSpread({
        borderRadius: radius,
        borderColor: interactiveColor,
        background: interactiveColor,
        color: whiteColor
      }, sizeNormal), {}, {
        selectors: {
          [`@media  only screen and (max-width: ${designtokenBreakpoints['ske-breakpoint-md']})`]: {
            width: props.mobileFullWidth ? '100%' : undefined
          }
        }
      });

    case 'warning':
      return _objectSpread(_objectSpread({
        borderRadius: radius,
        borderColor: designtokenColors['ske-color-burgundy-50'],
        background: designtokenColors['ske-color-burgundy-10'],
        color: designtokenColors['ske-color-black-100']
      }, sizeNormal), {}, {
        selectors: {
          [`@media  only screen and (max-width: ${designtokenBreakpoints['ske-breakpoint-md']})`]: {
            width: props.mobileFullWidth ? '100%' : 'auto'
          }
        }
      });

    case 'secondarySimple':
      return _objectSpread(_objectSpread({
        borderWidth: 3,
        borderRadius: radius,
        borderColor: 'transparent',
        background: 'none',
        color: interactiveColor
      }, sizeNormal), {}, {
        selectors: {
          [`@media  only screen and (max-width: ${designtokenBreakpoints['ske-breakpoint-md']})`]: {
            width: props.mobileFullWidth ? '100%' : 'auto'
          },
          '.ms-Button-textContainer': {
            textDecoration: 'underline'
          }
        }
      });

    default:
      // secondary
      return _objectSpread(_objectSpread({
        borderRadius: radius,
        borderColor: interactiveColor,
        background: whiteColor,
        color: interactiveColor
      }, sizeNormal), {}, {
        selectors: {
          [`@media  only screen and (max-width: ${designtokenBreakpoints['ske-breakpoint-md']})`]: {
            width: props.mobileFullWidth ? '100%' : 'auto'
          }
        }
      });
  }
}

function getTypeHoverColor(props) {
  const interactiveColor = designtokenColors['ske-color-interactive'];
  const interactiveDarkColor = designtokenColors['ske-color-interactive-dark'];
  const interactiveLightColor = designtokenColors['ske-color-interactive-light'];
  const errorColor = designtokenColors['ske-color-status-error'];
  const whiteColor = designtokenColors['ske-color-white-100'];

  switch (props.buttonStyle) {
    case 'warning':
      return {
        borderColor: errorColor,
        background: errorColor,
        color: whiteColor
      };

    case 'secondarySimple':
      return {
        borderColor: interactiveColor,
        background: interactiveLightColor,
        textDecoration: 'none',
        selectors: {
          '.ms-Button-textContainer': {
            textDecoration: 'none'
          }
        }
      };

    case 'primaryCornered':
    case 'primary':
    case 'callToAction':
      return {
        borderColor: interactiveDarkColor,
        background: interactiveDarkColor,
        color: whiteColor
      };

    default:
      return {
        borderColor: interactiveColor,
        background: interactiveLightColor,
        color: interactiveColor
      };
  }
}

function getDisabledColor(props) {
  switch (props.buttonStyle) {
    case 'primary':
      return {
        boxShadow: `none`
      };

    case 'secondary':
      return {
        background: 'none'
      };

    default:
      return;
  }
}

function setFocusRadius(props) {
  if (props.buttonStyle === 'primaryCornered') {
    return '8px';
  }

  if (props.buttonStyle === 'callToAction') {
    return '10px';
  } else {
    return '21px';
  }
}

export function getClassNames(props) {
  const theme = getTheme();
  const inset = props.buttonStyle === 'secondarySimple' ? -4 : -8;
  const disabled = props.disabled;
  const radius = setFocusRadius(props);
  return mergeStyles([getFocusStyle(theme, inset, 'relative', radius), {
    displayName: `ske-${props.buttonStyle}-button`,
    selectors: {
      '&.ms-Button': _objectSpread({
        borderWidth: '3px',
        fontWeight: 'normal',
        padding: '15px',
        transition: '0.2s',
        textAlign: 'center',
        verticalAlign: 'top'
      }, getTypeColor(props)),
      '&.ms-Button:hover': _objectSpread({}, getTypeHoverColor(props)),
      '&.ms-Button:active': {
        transition: '0.15s',
        boxShadow: 'none',
        transform: disabled ? 'none' : 'translateY(2px)',
        transitionTimingFunction: 'ease'
      },
      '&.ms-Button:disabled': _objectSpread({
        background: designtokenColors['ske-color-grey-10'],
        borderColor: designtokenColors['ske-color-grey-30'],
        color: designtokenColors['ske-color-grey-30'],
        cursor: 'not-allowed'
      }, getDisabledColor(props)),
      '&.ms-Button i': {
        fontSize: designtokenFontSizes['ske-font-size-icon-l']
      }
    }
  }]);
}
