// @skatteetaten/frontend-components/cjs/LabelWithCallout/LabelWithCallout.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _colors = _interopRequireDefault(require("../utils/designtokens/_colors.json"));

var _spacing = _interopRequireDefault(require("../utils/designtokens/_spacing.json"));

var _fontSizes = _interopRequireDefault(require("../utils/designtokens/_fontSizes.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function getLabelSize(props) {
  switch (props.inputSize) {
    case 'small':
      return {
        fontSize: _fontSizes.default['ske-font-size-xs']
      };

    case 'large':
      return {
        fontSize: _fontSizes.default['ske-font-size-icon-l']
      };

    default:
      return {
        fontSize: _fontSizes.default['ske-font-size-m']
      };
  }
}

function getDisplay(props) {
  const {
    help,
    warning
  } = props;
  return help !== undefined || warning !== undefined ? 'initial' : 'inline-block';
}

const getClassNames = props => {
  const {
    calloutFloating
  } = props;
  return (0, _mergeStyles.mergeStyleSets)({
    calloutLabelWrapper: {
      display: 'flex'
    },
    calloutContext: {
      selectors: {
        '& .ms-Callout-container': {
          position: !calloutFloating ? 'inherit' : undefined,
          margin: !calloutFloating ? `${_spacing.default['ske-spacing-lg']} 0` : 0,
          width: '100%'
        },
        '& .ms-Callout': !calloutFloating && {
          left: '0 !important',
          top: '0 !important',
          position: 'inherit'
        },
        '& .ms-Callout-main': !calloutFloating && {
          maxWidth: '100%',
          display: 'inline-block',
          overflow: 'initial'
        },
        '& .ms-Callout-beak': !calloutFloating && {
          left: `${_spacing.default['ske-spacing-xl']} !important`,
          top: `-${_spacing.default['ske-spacing-md']} !important`
        },
        '& .ms-Callout-beakCurtain': {
          backgroundColor: 'transparent'
        }
      }
    },
    label: _objectSpread(_objectSpread({
      paddingBottom: _spacing.default['ske-spacing-sm'],
      paddingLeft: 0,
      lineHeight: _fontSizes.default['ske-line-height-m'],
      display: getDisplay(props),
      color: _colors.default['ske-color-black-100'],
      fontWeight: _fontSizes.default['ske-font-weight-regular']
    }, getLabelSize(props)), {}, {
      selectors: {
        '& .ms-Label': {
          fontWeight: _fontSizes.default['ske-font-weight-regular']
        }
      }
    }),
    labelAsLegend: _objectSpread(_objectSpread({
      paddingLeft: 0,
      paddingBottom: _spacing.default['ske-spacing-sm'],
      display: 'inline-block',
      color: _colors.default['ske-color-black-100'],
      fontWeight: _fontSizes.default['ske-font-weight-regular']
    }, getLabelSize(props)), {}, {
      selectors: {
        '& .ms-Label': {
          fontWeight: _fontSizes.default['ske-font-weight-regular']
        },
        '@supports (display: contents)': {
          display: 'contents'
        }
      }
    }),
    labelIconArea: {
      height: _spacing.default['ske-spacing-xl'],
      display: 'inline-block',
      selectors: {
        '& button ': {
          height: _spacing.default['ske-spacing-xl'],
          width: _spacing.default['ske-spacing-xl'],
          marginLeft: _spacing.default['ske-spacing-xs'],
          borderRadius: '100%'
        }
      }
    },
    icon: {
      color: _colors.default['ske-color-interactive'],
      selectors: {
        '& i': {
          fontSize: _fontSizes.default['ske-font-size-l']
        },
        '&:hover': {
          background: _colors.default['ske-color-interactive-light']
        },
        '&:focus': {
          background: _colors.default['ske-color-white']
        },
        '&:focus&:after': {
          border: `2px solid ${_colors.default['ske-color-interactive']}`,
          borderRadius: '100%',
          inset: '-1px',
          outline: 'none'
        }
      }
    },
    warningicon: {
      color: _colors.default['ske-color-interactive'],
      selectors: {
        '& i': {
          fontSize: _fontSizes.default['ske-font-size-l']
        },
        '&:hover': {
          background: _colors.default['ske-color-interactive-light']
        },
        '&:focus': {
          background: _colors.default['ske-color-white']
        },
        '&:focus&:after': {
          border: `2px solid ${_colors.default['ske-color-interactive']}`,
          borderRadius: '100%',
          inset: '-1px',
          outline: 'none'
        }
      }
    },
    errorIcon: {
      position: 'absolute',
      left: `-${_spacing.default['ske-spacing-xl']}`,
      bottom: _spacing.default['ske-spacing-xxl'],
      selectors: {
        '& i': {
          color: _colors.default['ske-color-status-error'],
          fontSize: _fontSizes.default['ske-font-size-l']
        }
      }
    }
  });
};

exports.getClassNames = getClassNames;
