// @skatteetaten/frontend-components/cjs/entry.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Accordion", {
  enumerable: true,
  get: function () {
    return _Accordion.Accordion;
  }
});
Object.defineProperty(exports, "AccordionItem", {
  enumerable: true,
  get: function () {
    return _AccordionItem.AccordionItem;
  }
});
Object.defineProperty(exports, "AccordionMenu", {
  enumerable: true,
  get: function () {
    return _AccordionMenu.AccordionMenu;
  }
});
Object.defineProperty(exports, "AccordionMenuItem", {
  enumerable: true,
  get: function () {
    return _AccordionMenuItem.AccordionMenuItem;
  }
});
Object.defineProperty(exports, "ActionButton", {
  enumerable: true,
  get: function () {
    return _ActionButton.ActionButton;
  }
});
Object.defineProperty(exports, "Animation", {
  enumerable: true,
  get: function () {
    return _getAnimationStyles.Animation;
  }
});
Object.defineProperty(exports, "Article", {
  enumerable: true,
  get: function () {
    return _Article.Article;
  }
});
Object.defineProperty(exports, "Aside", {
  enumerable: true,
  get: function () {
    return _Aside.Aside;
  }
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function () {
    return _Button.Button;
  }
});
Object.defineProperty(exports, "ButtonLink", {
  enumerable: true,
  get: function () {
    return _ButtonLink.ButtonLink;
  }
});
Object.defineProperty(exports, "Callout", {
  enumerable: true,
  get: function () {
    return _Callout.Callout;
  }
});
Object.defineProperty(exports, "CalloutColor", {
  enumerable: true,
  get: function () {
    return _Callout2.CalloutColor;
  }
});
Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function () {
    return _Card.Card;
  }
});
Object.defineProperty(exports, "CardBorder", {
  enumerable: true,
  get: function () {
    return _Card2.CardBorder;
  }
});
Object.defineProperty(exports, "CardColor", {
  enumerable: true,
  get: function () {
    return _Card2.CardColor;
  }
});
Object.defineProperty(exports, "CheckBox", {
  enumerable: true,
  get: function () {
    return _CheckBox.CheckBox;
  }
});
Object.defineProperty(exports, "Chip", {
  enumerable: true,
  get: function () {
    return _Chip.Chip;
  }
});
Object.defineProperty(exports, "Col", {
  enumerable: true,
  get: function () {
    return _Col.Col;
  }
});
Object.defineProperty(exports, "ComboBox", {
  enumerable: true,
  get: function () {
    return _ComboBox.ComboBox;
  }
});
Object.defineProperty(exports, "DatePicker", {
  enumerable: true,
  get: function () {
    return _DatePicker.DatePicker;
  }
});
Object.defineProperty(exports, "DayOfWeek", {
  enumerable: true,
  get: function () {
    return _DatePicker.DayOfWeek;
  }
});
Object.defineProperty(exports, "DetailsList", {
  enumerable: true,
  get: function () {
    return _DetailsList.DetailsList;
  }
});
Object.defineProperty(exports, "DetailsRow", {
  enumerable: true,
  get: function () {
    return _DetailsList.DetailsRow;
  }
});
Object.defineProperty(exports, "Dialog", {
  enumerable: true,
  get: function () {
    return _Dialog.Dialog;
  }
});
Object.defineProperty(exports, "Dropdown", {
  enumerable: true,
  get: function () {
    return _Dropdown.Dropdown;
  }
});
Object.defineProperty(exports, "ErrorMessage", {
  enumerable: true,
  get: function () {
    return _ErrorMessage.ErrorMessage;
  }
});
Object.defineProperty(exports, "ErrorSummary", {
  enumerable: true,
  get: function () {
    return _ErrorSummary.ErrorSummary;
  }
});
Object.defineProperty(exports, "ExternalHeader", {
  enumerable: true,
  get: function () {
    return _TopBanner.ExternalHeader;
  }
});
Object.defineProperty(exports, "FileFormatTypes", {
  enumerable: true,
  get: function () {
    return _FileUploader2.FileFormatTypes;
  }
});
Object.defineProperty(exports, "FileUploader", {
  enumerable: true,
  get: function () {
    return _FileUploader.FileUploader;
  }
});
Object.defineProperty(exports, "FirstWeekOfYear", {
  enumerable: true,
  get: function () {
    return _DatePicker.FirstWeekOfYear;
  }
});
Object.defineProperty(exports, "FontSizes", {
  enumerable: true,
  get: function () {
    return _fonts.FontSizes;
  }
});
Object.defineProperty(exports, "FontWeights", {
  enumerable: true,
  get: function () {
    return _fonts.FontWeights;
  }
});
Object.defineProperty(exports, "Fonts", {
  enumerable: true,
  get: function () {
    return _fonts.Fonts;
  }
});
Object.defineProperty(exports, "Footer", {
  enumerable: true,
  get: function () {
    return _Footer.Footer;
  }
});
Object.defineProperty(exports, "FooterContent", {
  enumerable: true,
  get: function () {
    return _FooterContent.FooterContent;
  }
});
Object.defineProperty(exports, "Grid", {
  enumerable: true,
  get: function () {
    return _Grid.Grid;
  }
});
Object.defineProperty(exports, "Header", {
  enumerable: true,
  get: function () {
    return _Header.Header;
  }
});
Object.defineProperty(exports, "Heading", {
  enumerable: true,
  get: function () {
    return _Heading.Heading;
  }
});
Object.defineProperty(exports, "Icon", {
  enumerable: true,
  get: function () {
    return _Icon.Icon;
  }
});
Object.defineProperty(exports, "IconButton", {
  enumerable: true,
  get: function () {
    return _IconButton.IconButton;
  }
});
Object.defineProperty(exports, "IconFontSizes", {
  enumerable: true,
  get: function () {
    return _fonts.IconFontSizes;
  }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function () {
    return _Image.Image;
  }
});
Object.defineProperty(exports, "LabelWithCallout", {
  enumerable: true,
  get: function () {
    return _LabelWithCallout.LabelWithCallout;
  }
});
Object.defineProperty(exports, "Language", {
  enumerable: true,
  get: function () {
    return _Language.Language;
  }
});
Object.defineProperty(exports, "LanguagePicker", {
  enumerable: true,
  get: function () {
    return _LanguagePicker.LanguagePicker;
  }
});
Object.defineProperty(exports, "Layout", {
  enumerable: true,
  get: function () {
    return _Layout.Layout;
  }
});
Object.defineProperty(exports, "LineHeightSizes", {
  enumerable: true,
  get: function () {
    return _fonts.LineHeightSizes;
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function () {
    return _Link.Link;
  }
});
Object.defineProperty(exports, "LinkGroup", {
  enumerable: true,
  get: function () {
    return _LinkGroup.LinkGroup;
  }
});
Object.defineProperty(exports, "Main", {
  enumerable: true,
  get: function () {
    return _Main.Main;
  }
});
Object.defineProperty(exports, "MessageBar", {
  enumerable: true,
  get: function () {
    return _MessageBar.MessageBar;
  }
});
Object.defineProperty(exports, "Nav", {
  enumerable: true,
  get: function () {
    return _Nav.Nav;
  }
});
Object.defineProperty(exports, "NavigationContent", {
  enumerable: true,
  get: function () {
    return _NavigationContent.NavigationContent;
  }
});
Object.defineProperty(exports, "NavigationTile", {
  enumerable: true,
  get: function () {
    return _NavigationTile.NavigationTile;
  }
});
Object.defineProperty(exports, "NextPage", {
  enumerable: true,
  get: function () {
    return _Pagination.NextPage;
  }
});
Object.defineProperty(exports, "OpenClose", {
  enumerable: true,
  get: function () {
    return _OpenClose.OpenClose;
  }
});
Object.defineProperty(exports, "Page", {
  enumerable: true,
  get: function () {
    return _Pagination.Page;
  }
});
Object.defineProperty(exports, "Pages", {
  enumerable: true,
  get: function () {
    return _Pagination.Pages;
  }
});
Object.defineProperty(exports, "Pagination", {
  enumerable: true,
  get: function () {
    return _Pagination.Pagination;
  }
});
Object.defineProperty(exports, "Palette", {
  enumerable: true,
  get: function () {
    return _palette.DefaultPalette;
  }
});
Object.defineProperty(exports, "PreviousLink", {
  enumerable: true,
  get: function () {
    return _Pagination.PreviousLink;
  }
});
Object.defineProperty(exports, "ProgressBar", {
  enumerable: true,
  get: function () {
    return _ProgressBar.ProgressBar;
  }
});
Object.defineProperty(exports, "RadioButtonGroup", {
  enumerable: true,
  get: function () {
    return _RadioButtonGroup.RadioButtonGroup;
  }
});
Object.defineProperty(exports, "Row", {
  enumerable: true,
  get: function () {
    return _Row.Row;
  }
});
Object.defineProperty(exports, "ScrollToTopButton", {
  enumerable: true,
  get: function () {
    return _ScrollToTopButton.ScrollToTopButton;
  }
});
Object.defineProperty(exports, "SearchField", {
  enumerable: true,
  get: function () {
    return _SearchField.SearchField;
  }
});
Object.defineProperty(exports, "SkeBasis", {
  enumerable: true,
  get: function () {
    return _SkeBasis.SkeBasis;
  }
});
Object.defineProperty(exports, "SkeIcons", {
  enumerable: true,
  get: function () {
    return _iconsSke.default;
  }
});
Object.defineProperty(exports, "Spinner", {
  enumerable: true,
  get: function () {
    return _Spinner.Spinner;
  }
});
Object.defineProperty(exports, "Step", {
  enumerable: true,
  get: function () {
    return _Step.Step;
  }
});
Object.defineProperty(exports, "StepList", {
  enumerable: true,
  get: function () {
    return _StepList.StepList;
  }
});
Object.defineProperty(exports, "TabItem", {
  enumerable: true,
  get: function () {
    return _TabItem.TabItem;
  }
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function () {
    return _Table.Table;
  }
});
Object.defineProperty(exports, "TableRow", {
  enumerable: true,
  get: function () {
    return _TableRow.TableRow;
  }
});
Object.defineProperty(exports, "Tabs", {
  enumerable: true,
  get: function () {
    return _Tabs.Tabs;
  }
});
Object.defineProperty(exports, "TextField", {
  enumerable: true,
  get: function () {
    return _TextField.TextField;
  }
});
Object.defineProperty(exports, "TopBanner", {
  enumerable: true,
  get: function () {
    return _TopBanner.TopBanner;
  }
});
Object.defineProperty(exports, "TopStripe", {
  enumerable: true,
  get: function () {
    return _TopStripe.TopStripe;
  }
});
Object.defineProperty(exports, "TopStripeButton", {
  enumerable: true,
  get: function () {
    return _TopStripeButton.TopStripeButton;
  }
});
Object.defineProperty(exports, "TopStripeContext", {
  enumerable: true,
  get: function () {
    return _TopStripe.TopStripeContext;
  }
});
Object.defineProperty(exports, "TopStripeMenu", {
  enumerable: true,
  get: function () {
    return _TopStripeMenu.TopStripeMenu;
  }
});
Object.defineProperty(exports, "Typography", {
  enumerable: true,
  get: function () {
    return _Typography.Typography;
  }
});
Object.defineProperty(exports, "UseScreen", {
  enumerable: true,
  get: function () {
    return _ScreenPlugin.UseScreen;
  }
});
Object.defineProperty(exports, "createFont", {
  enumerable: true,
  get: function () {
    return _fonts.createFont;
  }
});
Object.defineProperty(exports, "generateId", {
  enumerable: true,
  get: function () {
    return _generateId.generateId;
  }
});
Object.defineProperty(exports, "getFocusStyle", {
  enumerable: true,
  get: function () {
    return _getFocusStyle.getFocusStyle;
  }
});
Object.defineProperty(exports, "getSlidingWindowEdges", {
  enumerable: true,
  get: function () {
    return _Pagination.getSlidingWindowEdges;
  }
});
Object.defineProperty(exports, "getSrOnlyStyle", {
  enumerable: true,
  get: function () {
    return _getSrOnlyStyle.getSrOnlyStyle;
  }
});
Object.defineProperty(exports, "languages", {
  enumerable: true,
  get: function () {
    return _languages.languages;
  }
});
Object.defineProperty(exports, "omstrukturerFlatJSON", {
  enumerable: true,
  get: function () {
    return _i18n.omstrukturerFlatJSON;
  }
});
Object.defineProperty(exports, "skeColor", {
  enumerable: true,
  get: function () {
    return _palette.skeColor;
  }
});
Object.defineProperty(exports, "skePalette", {
  enumerable: true,
  get: function () {
    return _palette.skePalette;
  }
});
Object.defineProperty(exports, "t", {
  enumerable: true,
  get: function () {
    return _i18n.t;
  }
});
Object.defineProperty(exports, "takeIf", {
  enumerable: true,
  get: function () {
    return _helpers.takeIf;
  }
});
Object.defineProperty(exports, "useHotkeys", {
  enumerable: true,
  get: function () {
    return _useHotkeys.useHotkeys;
  }
});

var _palette = require("./utils/palette");

var _fonts = require("./utils/fonts");

var _i18n = require("./utils/i18n/i18n");

var _getSrOnlyStyle = require("./utils/getSrOnlyStyle");

var _languages = require("./utils/i18n/languages");

var _Language = require("./utils/Language");

var _getFocusStyle = require("./utils/getFocusStyle");

var _Heading = require("./utils/Heading");

var _ScreenPlugin = require("./utils/ScreenPlugin");

var _iconsSke = _interopRequireDefault(require("./utils/icons/icons-ske"));

var _getAnimationStyles = require("./utils/getAnimationStyles");

var _helpers = require("./utils/helpers");

var _generateId = require("./utils/generateId");

var _useHotkeys = require("./utils/useHotkeys");

var _Accordion = require("./Accordion/Accordion");

var _AccordionItem = require("./Accordion/AccordionItem/AccordionItem");

var _AccordionMenu = require("./AccordionMenu/AccordionMenu");

var _AccordionMenuItem = require("./AccordionMenu/AccordionMenuItem/AccordionMenuItem");

var _ActionButton = require("./ActionButton/ActionButton");

var _Button = require("./Button/Button");

var _ButtonLink = require("./ButtonLink/ButtonLink");

var _Callout = require("./Callout/Callout");

var _Callout2 = require("./Callout/Callout.types");

var _Card = require("./Card/Card");

var _Card2 = require("./Card/Card.types");

var _CheckBox = require("./CheckBox/CheckBox");

var _Chip = require("./Chip/Chip");

var _ComboBox = require("./ComboBox/ComboBox");

var _DatePicker = require("./DatePicker/DatePicker");

var _DetailsList = require("./DetailsList/DetailsList");

var _Dialog = require("./Dialog/Dialog");

var _Dropdown = require("./Dropdown/Dropdown");

var _ErrorMessage = require("./ErrorMessage/ErrorMessage");

var _ErrorSummary = require("./ErrorSummary/ErrorSummary");

var _FileUploader = require("./FileUploader/FileUploader");

var _FileUploader2 = require("./FileUploader/FileUploader.types");

var _FooterContent = require("./FooterContent/FooterContent");

var _Grid = require("./Grid/Grid");

var _Row = require("./Grid/Row");

var _Col = require("./Grid/Col");

var _Icon = require("./Icon/Icon");

var _IconButton = require("./IconButton/IconButton");

var _Image = require("./Image/Image");

var _LabelWithCallout = require("./LabelWithCallout/LabelWithCallout");

var _Link = require("./Link/Link");

var _LinkGroup = require("./LinkGroup/LinkGroup");

var _MessageBar = require("./MessageBar/MessageBar");

var _NavigationTile = require("./NavigationTile/NavigationTile");

var _NavigationContent = require("./NavigationTile/NavigationContent/NavigationContent");

var _OpenClose = require("./OpenClose/OpenClose");

var _Pagination = require("./Pagination/Pagination");

var _ProgressBar = require("./ProgressBar/ProgressBar");

var _RadioButtonGroup = require("./RadioButtonGroup/RadioButtonGroup");

var _ScrollToTopButton = require("./ScrollToTopButton/ScrollToTopButton");

var _SearchField = require("./SearchField/SearchField");

var _SkeBasis = require("./SkeBasis/SkeBasis");

var _Spinner = require("./Spinner/Spinner");

var _StepList = require("./StepList/StepList");

var _Step = require("./StepList/Step/Step");

var _Table = require("./Table/Table");

var _TableRow = require("./Table/TableRow/TableRow");

var _Tabs = require("./Tabs/Tabs");

var _TabItem = require("./Tabs/TabItem/TabItem");

var _TextField = require("./TextField/TextField");

var _TopBanner = require("./TopBanner/TopBanner");

var _TopStripe = require("./TopStripe/TopStripe");

var _TopStripeButton = require("./TopStripe/TopStripeButton");

var _TopStripeMenu = require("./TopStripe/TopStripeMenu");

var _LanguagePicker = require("./TopStripe/LanguagePicker");

var _Typography = require("./Typography/Typography");

var _Layout = require("./Layout/Layout");

var _Article = require("./Layout/Article");

var _Aside = require("./Layout/Aside");

var _Footer = require("./Layout/Footer");

var _Header = require("./Layout/Header");

var _Main = require("./Layout/Main");

var _Nav = require("./Layout/Nav");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
