// @skatteetaten/frontend-components/cjs/Accordion/Accordion.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getClassNames = getClassNames;

var _mergeStyles = require("@fluentui/merge-styles");

var _breakpoints = _interopRequireDefault(require("../utils/designtokens/_breakpoints.json"));

var _spacing = _interopRequireDefault(require("../utils/designtokens/_spacing.json"));

var _fontSizes = _interopRequireDefault(require("../utils/designtokens/_fontSizes.json"));

var _colors = _interopRequireDefault(require("../utils/designtokens/_colors.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getClassNames() {
  const paddingAccordionContent = `0 0 ${_spacing.default['ske-spacing-lg']} ${_spacing.default['ske-spacing-md']}`;
  return (0, _mergeStyles.mergeStyleSets)({
    containerStep: {
      margin: '0 auto'
    },
    wrapperStep: {
      position: 'relative'
    },
    accordion: {
      position: 'relative',
      backgroundColor: _colors.default['ske-color-white-100'],
      marginBottom: _spacing.default['ske-spacing-lg'],
      selectors: {
        '& hr': {
          margin: 0,
          border: 'solid',
          borderWidth: '1px',
          color: _colors.default['ske-color-grey-30']
        }
      }
    },
    content: {
      backgroundColor: _colors.default['ske-color-white-100'],
      padding: paddingAccordionContent,
      selectors: {
        '&:focus': {
          background: _colors.default['ske-color-blue-10']
        }
      }
    },
    heading: {
      fontSize: _fontSizes.default['ske-font-size-icon-l'],
      margin: '0'
    },
    stepNumber: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: _colors.default['ske-color-white-100'],
      border: 'solid',
      width: _spacing.default['ske-spacing-xxl'],
      height: _spacing.default['ske-spacing-xxl'],
      borderRadius: '50%',
      borderWidth: '2px',
      borderColor: _colors.default['ske-color-black-100'],
      fontSize: _fontSizes.default['ske-font-size-l'],
      fontWeight: _fontSizes.default['ske-font-weight-bold'],
      zIndex: 10,
      marginTop: _spacing.default['ske-spacing-lg'],
      selectors: {
        [`@media (max-width: ${_breakpoints.default['ske-breakpoint-md']})`]: {
          fontSize: _fontSizes.default['ske-font-size-m'],
          width: _spacing.default['ske-spacing-xl'],
          height: _spacing.default['ske-spacing-xl']
        }
      }
    },
    stepLine: {
      display: 'block',
      position: 'absolute',
      width: 2,
      backgroundColor: _colors.default['ske-color-grey-30'],
      top: _spacing.default['ske-spacing-xxl'],
      left: '0.8rem',
      height: '100%',
      selectors: {
        [`@media (min-width: ${_breakpoints.default['ske-breakpoint-md']})`]: {
          top: _spacing.default['ske-spacing-xl'],
          left: _spacing.default['ske-spacing-lg'],
          marginLeft: 1
        }
      }
    },
    toggleButton: {
      width: '100%',
      border: 'none',
      textAlign: 'left',
      color: _colors.default['ske-color-interactive'],
      fontSize: _fontSizes.default['ske-font-size-m'],
      fontWeight: _fontSizes.default['ske-font-weight-bold'],
      padding: `${_spacing.default['ske-spacing-lg']} ${_spacing.default['ske-spacing-lg']} ${_spacing.default['ske-spacing-lg']} ${_spacing.default['ske-spacing-md']}`,
      background: 'none',
      position: 'relative',
      cursor: 'pointer',
      marginTop: 0,
      marginBottom: 0,
      selectors: {
        [`@media (min-width: ${_breakpoints.default['ske-breakpoint-md']})`]: {
          fontSize: _fontSizes.default['ske-font-size-icon-l']
        },
        '&:hover, &:focus': {
          background: _colors.default['ske-color-interactive-light']
        },
        '&:focus': {
          outline: 'none'
        },
        '& h1, h2, h3, h4, h5, h6': {
          fontSize: _fontSizes.default['ske-font-size-icon-l'],
          margin: '0'
        },
        '& i': {
          alignSelf: 'flex-start',
          transition: '.2s',
          marginTop: '-2px',
          fontSize: _fontSizes.default['ske-font-size-icon-xl'],
          selectors: {
            '@media (min-width: 640px)': {
              fontSize: _fontSizes.default['ske-font-size-icon-xxl']
            }
          }
        }
      }
    },
    toggleButtonOpen: {
      selectors: {
        '& i': {
          transform: 'rotate(180deg)'
        }
      }
    },
    toggleButtonContent: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center'
    },
    subtitle: {
      display: 'block',
      color: _colors.default['ske-color-black-100'],
      width: '100%',
      border: 'none',
      textAlign: 'left',
      fontSize: _fontSizes.default['ske-font-size-l'],
      fontWeight: _fontSizes.default['ske-font-weight-regular'],
      margin: 0,
      padding: `${_spacing.default['ske-spacing-md']} ${_spacing.default['ske-spacing-sm']} 0 0`
    }
  });
}
