// @skatteetaten/frontend-components/cjs/FooterContent/FooterContent.classNames.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getLogoClassNames = exports.getClassNames = void 0;

var _mergeStyles = require("@fluentui/merge-styles");

var _react = require("@fluentui/react");

const getClassNames = () => {
  const palette = (0, _react.getTheme)().palette;
  return (0, _mergeStyles.mergeStyleSets)({
    footerWrapperSKE: {
      backgroundColor: palette.skeColor.burgundy,
      color: palette.skeColor.white
    },
    footerContentSKE: {
      backgroundColor: palette.skeColor.burgundy,
      padding: '24px 4px',
      selectors: {
        '@media (max-width: 1023px)': {
          maxWidth: '100%',
          padding: '24px 16px'
        }
      }
    },

    /** BRAND VARIANTS INK */
    footerWrapperINK: {
      backgroundColor: palette.skeColor.green100,
      color: palette.skeColor.white
    },
    footerContentINK: {
      backgroundColor: palette.skeColor.green100,
      padding: '24px 4px',
      selectors: {
        '@media (max-width: 1023px)': {
          maxWidth: '100%',
          padding: '24px 16px'
        }
      }
    },

    /** BRAND VARIANTS LSO */
    footerWrapperLSO: {
      backgroundColor: palette.skeColor.black100,
      color: palette.skeColor.white
    },
    footerContentLSO: {
      backgroundColor: palette.skeColor.black100,
      padding: '24px 4px',
      selectors: {
        '@media (max-width: 1023px)': {
          maxWidth: '100%',
          padding: '24px 16px'
        }
      }
    },
    footerDecorContainer: {
      width: '100%',
      marginBottom: -5,
      lineHeight: 0,
      selectors: {
        svg: {
          height: '85px',
          overflow: 'hidden'
        }
      }
    }
  });
};

exports.getClassNames = getClassNames;

const getLogoClassNames = () => (0, _mergeStyles.mergeStyles)([{
  selectors: {
    '@media (max-width: 1023px)': {
      selectors: {
        img: {
          margin: '0 auto'
        },
        'div &': {
          marginBottom: '16px'
        }
      }
    }
  }
}]);

exports.getLogoClassNames = getLogoClassNames;
