// @skatteetaten/frontend-components/cjs/FooterContent/footerDekor.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = (primaryColor = '#6F2C3F', secondaryColor = '#F3A7B0') => /*#__PURE__*/_react.default.createElement("svg", {
  focusable: "false",
  className: "footer-decoration-svg",
  xmlns: "http://www.w3.org/2000/svg",
  width: "100%",
  height: "100%",
  preserveAspectRatio: "none",
  viewBox: "0 0 1440 100",
  version: "1.1"
}, /*#__PURE__*/_react.default.createElement("desc", null), /*#__PURE__*/_react.default.createElement("defs", null), /*#__PURE__*/_react.default.createElement("g", {
  stroke: "none",
  strokeWidth: "0",
  fill: "none",
  fillRule: "evenodd"
}, /*#__PURE__*/_react.default.createElement("g", {
  id: "bakgrunn"
}, /*#__PURE__*/_react.default.createElement("polygon", {
  fill: secondaryColor,
  points: "0 99.6400524 1440 99.6400524 1440 69.2255838 0 17.1793194"
}), /*#__PURE__*/_react.default.createElement("path", {
  d: "M1440,98 L1440,629 L0,629 L0,98 L1440,0 L1440,98 Z",
  fill: primaryColor
}))));

exports.default = _default;
