// @skatteetaten/frontend-components/cjs/FooterContent/FooterContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FooterContent = void 0;

var React = _interopRequireWildcard(require("react"));

var _Image = require("../Image");

var _FooterContent = require("./FooterContent.classNames");

var _footerDekor = _interopRequireDefault(require("./footerDekor"));

var _skeLogo = _interopRequireDefault(require("./assets/ske-logo.svg"));

var _skeLogoEn = _interopRequireDefault(require("./assets/ske-logo-en.svg"));

var _inkLogo = _interopRequireDefault(require("./assets/ink-logo.svg"));

var _SkeBasis = require("../SkeBasis");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

const Logo = () => {
  return /*#__PURE__*/React.createElement(_Image.Image, {
    src: _skeLogo.default,
    className: (0, _FooterContent.getLogoClassNames)(),
    height: "74px",
    alt: "Skatteetaten logo"
  });
};

const LogoEn = () => {
  return /*#__PURE__*/React.createElement(_Image.Image, {
    src: _skeLogoEn.default,
    className: (0, _FooterContent.getLogoClassNames)(),
    height: "74px",
    alt: "Norwegian Tax Authorities logo"
  });
};

const LogoInk = () => {
  return /*#__PURE__*/React.createElement(_Image.Image, {
    src: _inkLogo.default,
    className: (0, _FooterContent.getLogoClassNames)(),
    height: "63px",
    alt: "Statens innkreving - en del av Skatteetaten, logo"
  });
};
/*
 * visibleName FooterContent (Bunn)
 */


class FooterContent extends React.PureComponent {
  /** Engelsk logo */

  /** Norsk Logo */

  /** Statens innkreving logo */
  render() {
    const {
      children,
      className,
      ariaLabel
    } = this.props;
    const styles = (0, _FooterContent.getClassNames)();
    return /*#__PURE__*/React.createElement(_SkeBasis.BrandContext.Consumer, null, ({
      tag,
      primaryColor,
      secondaryColor
    }) => /*#__PURE__*/React.createElement("div", {
      className: className,
      "aria-label": ariaLabel
    }, /*#__PURE__*/React.createElement("div", {
      className: styles.footerDecorContainer,
      "aria-hidden": "true"
    }, (0, _footerDekor.default)(primaryColor, secondaryColor)), /*#__PURE__*/React.createElement("footer", {
      className: styles['footerWrapper' + tag]
    }, /*#__PURE__*/React.createElement("div", {
      className: styles['footerContent' + tag]
    }, children))));
  }

}

exports.FooterContent = FooterContent;

_defineProperty(FooterContent, "LogoEn", LogoEn);

_defineProperty(FooterContent, "Logo", Logo);

_defineProperty(FooterContent, "LogoInk", LogoInk);
