// @skatteetaten/frontend-components/cjs/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AliasIcons", {
  enumerable: true,
  get: function () {
    return _iconsAlias.default;
  }
});
Object.defineProperty(exports, "Animation", {
  enumerable: true,
  get: function () {
    return _getAnimationStyles.Animation;
  }
});
Object.defineProperty(exports, "FontSizes", {
  enumerable: true,
  get: function () {
    return _fonts.FontSizes;
  }
});
Object.defineProperty(exports, "FontWeights", {
  enumerable: true,
  get: function () {
    return _fonts.FontWeights;
  }
});
Object.defineProperty(exports, "Fonts", {
  enumerable: true,
  get: function () {
    return _fonts.Fonts;
  }
});
Object.defineProperty(exports, "Heading", {
  enumerable: true,
  get: function () {
    return _Heading.Heading;
  }
});
Object.defineProperty(exports, "IconFontSizes", {
  enumerable: true,
  get: function () {
    return _fonts.IconFontSizes;
  }
});
Object.defineProperty(exports, "LineHeightSizes", {
  enumerable: true,
  get: function () {
    return _fonts.LineHeightSizes;
  }
});
Object.defineProperty(exports, "Palette", {
  enumerable: true,
  get: function () {
    return _palette.DefaultPalette;
  }
});
Object.defineProperty(exports, "SkeIcons", {
  enumerable: true,
  get: function () {
    return _iconsSke.default;
  }
});
Object.defineProperty(exports, "UseScreen", {
  enumerable: true,
  get: function () {
    return _ScreenPlugin.UseScreen;
  }
});
Object.defineProperty(exports, "createFont", {
  enumerable: true,
  get: function () {
    return _fonts.createFont;
  }
});
Object.defineProperty(exports, "generateId", {
  enumerable: true,
  get: function () {
    return _generateId.generateId;
  }
});
Object.defineProperty(exports, "getFocusStyle", {
  enumerable: true,
  get: function () {
    return _getFocusStyle.getFocusStyle;
  }
});
Object.defineProperty(exports, "getSrOnlyStyle", {
  enumerable: true,
  get: function () {
    return _getSrOnlyStyle.getSrOnlyStyle;
  }
});
Object.defineProperty(exports, "languages", {
  enumerable: true,
  get: function () {
    return _languages.languages;
  }
});
Object.defineProperty(exports, "omstrukturerFlatJSON", {
  enumerable: true,
  get: function () {
    return _i18n.omstrukturerFlatJSON;
  }
});
Object.defineProperty(exports, "skeColor", {
  enumerable: true,
  get: function () {
    return _palette.skeColor;
  }
});
Object.defineProperty(exports, "skePalette", {
  enumerable: true,
  get: function () {
    return _palette.skePalette;
  }
});
Object.defineProperty(exports, "t", {
  enumerable: true,
  get: function () {
    return _i18n.t;
  }
});
Object.defineProperty(exports, "takeIf", {
  enumerable: true,
  get: function () {
    return _helpers.takeIf;
  }
});
Object.defineProperty(exports, "useHotkeys", {
  enumerable: true,
  get: function () {
    return _useHotkeys.useHotkeys;
  }
});

var _palette = require("./palette");

var _fonts = require("./fonts");

var _i18n = require("./i18n/i18n");

var _getSrOnlyStyle = require("./getSrOnlyStyle");

var _languages = require("./i18n/languages");

var _getFocusStyle = require("./getFocusStyle");

var _Heading = require("./Heading");

var _ScreenPlugin = require("./ScreenPlugin");

var _iconsSke = _interopRequireDefault(require("./icons/icons-ske"));

var _iconsAlias = _interopRequireDefault(require("./icons/icons-alias"));

var _getAnimationStyles = require("./getAnimationStyles");

var _helpers = require("./helpers");

var _generateId = require("./generateId");

var _useHotkeys = require("./useHotkeys");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
