// @skatteetaten/frontend-components/cjs/utils/palette.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.skePalette = exports.skeColor = exports.DefaultPalette = void 0;

var _colors = _interopRequireDefault(require("./designtokens/_colors.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const skeColor = {
  burgundy100: _colors.default['ske-color-burgundy-100'],
  burgundy70: _colors.default['ske-color-burgundy-70'],
  burgundy50: _colors.default['ske-color-burgundy-50'],
  burgundy30: _colors.default['ske-color-burgundy-30'],
  burgundy10: _colors.default['ske-color-burgundy-10'],
  burgundy5: _colors.default['ske-color-burgundy-5'],
  green100: _colors.default['ske-color-green-100'],
  green70: _colors.default['ske-color-green-70'],
  green50: _colors.default['ske-color-green-50'],
  green30: _colors.default['ske-color-green-30'],
  green10: _colors.default['ske-color-green-10'],
  green5: _colors.default['ske-color-green-5'],
  brown100: _colors.default['ske-color-brown-100'],
  brown70: _colors.default['ske-color-brown-70'],
  brown50: _colors.default['ske-color-brown-50'],
  brown30: _colors.default['ske-color-brown-30'],
  brown10: _colors.default['ske-color-brown-10'],
  brown5: _colors.default['ske-color-brown-5'],
  blue100: _colors.default['ske-color-blue-100'],
  blue70: _colors.default['ske-color-blue-70'],
  blue50: _colors.default['ske-color-blue-50'],
  blue30: _colors.default['ske-color-blue-30'],
  blue10: _colors.default['ske-color-blue-10'],
  blue5: _colors.default['ske-color-blue-5'],
  black100: _colors.default['ske-color-black-100'],
  grey70: _colors.default['ske-color-grey-70'],
  grey50: _colors.default['ske-color-grey-50'],
  grey30: _colors.default['ske-color-grey-30'],
  grey10: _colors.default['ske-color-grey-10'],
  grey5: _colors.default['ske-color-grey-5'],
  statusError: _colors.default['ske-color-status-error'],
  statusWarning: _colors.default['ske-color-status-warning'],
  statusOk: _colors.default['ske-color-status-ok'],
  interactiveDark: _colors.default['ske-color-interactive-dark'],
  interactive: _colors.default['ske-color-interactive'],
  interactiveLight: _colors.default['ske-color-interactive-light'],
  //ALIASES
  backgroundHoverColor: _colors.default['ske-color-blue-10'],
  backgroundFocusColor: _colors.default['ske-color-blue-10'],
  internal: _colors.default['ske-color-burgundy-100'],
  internalLight: _colors.default['ske-color-burgundy-5'],
  black: '#000000',
  blackAlt: _colors.default['ske-color-black-100'],
  darkGrey: _colors.default['ske-color-grey-70'],
  grey: _colors.default['ske-color-grey-50'],
  lightGrey: _colors.default['ske-color-grey-30'],
  neutralGrey: _colors.default['ske-color-grey-5'],
  whiteGrey: _colors.default['ske-color-grey-10'],
  white: '#ffffff',
  burgundy: _colors.default['ske-color-burgundy-100'],
  burgundyLight: _colors.default['ske-color-burgundy-70'],
  pink: '#df4661'
  /** UTGÅR */
  ,
  lightPink: '#f7d1d7'
  /** UTGÅR */
  ,
  brown: _colors.default['ske-color-brown-50'],
  lightBrown: _colors.default['ske-color-brown-30']
  /** Sjekk */
  ,
  beige: _colors.default['ske-color-brown-10'],
  lightBeige: _colors.default['ske-color-brown-5'],
  darkBeige: _colors.default['ske-color-brown-30'],
  green: _colors.default['ske-color-green-50'],
  lightGreen: _colors.default['ske-color-green-10'],
  darkBlue: _colors.default['ske-color-interactive-dark'],
  blue: _colors.default['ske-color-interactive'],
  lightBlue: _colors.default['ske-color-interactive-light'],
  error: _colors.default['ske-color-status-error'],
  transparent: 'transparent'
};
exports.skeColor = skeColor;
const skePalette = {
  //Sets up fluentUI baseline
  themePrimary: _colors.default['ske-color-interactive'],
  themeLighterAlt: '#ebf4fd',
  themeLighter: '#cce3f9',
  themeLight: '#82bcf1',
  themeTertiary: '#288be7',
  themeSecondary: '#146aba',
  themeDarkAlt: '#11599c',
  themeDark: '#0c3f6f',
  themeDarker: '#0a365f',
  neutralLighterAlt: '#f8f8f8',
  neutralLighter: _colors.default['ske-color-grey-5'],
  neutralLight: '#eaeaea',
  neutralQuaternaryAlt: '#dadada',
  neutralQuaternary: '#d0d0d0',
  neutralTertiaryAlt: '#c8c8c8',
  neutralTertiary: 'rgba(89, 89, 89, 0.95)',
  neutralSecondary: 'rgba(55, 55, 55, 0.95)',
  neutralPrimaryAlt: 'rgba(47, 47, 47, 0.95)',
  neutralPrimary: 'rgba(0, 0, 0, 0.95)',
  neutralDark: 'rgba(21, 21, 21, 0.95)',
  black: _colors.default['ske-color-black-100'],
  white: '#ffffff',
  primaryBackground: '#ffffff',
  primaryText: _colors.default['ske-color-black-100'],
  bodyBackground: '#ffffff',
  bodyText: 'rgba(0, 0, 0, 0.95)',
  disabledBackground: _colors.default['ske-color-grey-5'],
  disabledText: _colors.default['ske-color-grey-50']
}; //@ts-ignore

exports.skePalette = skePalette;
const DefaultPalette = Object.assign(skePalette, {
  // Egendefinerte farger fra Skatteetatens designmal
  skeColor: skeColor
  /* OPPRINNELIG LISTE
  themeDarker: '#004578',
  themeDark: '#005a9e',
  themeDarkAlt: '#106ebe',
  themePrimary: '#1362AE',
  themeSecondary: '#2b88d8',
  themeTertiary: '#71afe5',
  themeLight: '#c7e0f4',
  themeLighter: '#deecf9',
  themeLighterAlt: '#eff6fc',
  black: '#000000',
  blackTranslucent40: 'rgba(0,0,0,.4)',
  neutralDark: '#212121',
  neutralPrimary: '#333333',
  neutralPrimaryAlt: '#3c3c3c',
  neutralSecondary: '#666666',
  neutralTertiary: '#a6a6a6',
  neutralTertiaryAlt: '#c8c8c8',
  neutralQuaternary: '#d0d0d0',
  neutralQuaternaryAlt: '#dadada',
  neutralLight: '#eaeaea',
  neutralLighter: '#f4f4f4',
  neutralLighterAlt: '#f8f8f8',
  accent: '#0078d7',
  white: '#ffffff',
  yellow: '#ffb900',
  yellowLight: '#fff100',
  orange: '#d83b01',
  orangeLight: '#ea4300',
  orangeLighter: '#ff8c00',
  redDark: '#B70202',
  red: '#e81123',
  magentaDark: '#5c005c',
  magenta: '#b4009e',
  magentaLight: '#e3008c',
  purpleDark: '#32145a',
  purple: '#5c2d91',
  purpleLight: '#b4a0ff',
  blueDark: '#002050',
  blueMid: '#00188f',
  blue: '#0078d7',
  blueLight: '#00bcf2',
  tealDark: '#004b50',
  teal: '#008272',
  tealLight: '#00b294',
  greenDark: '#004b1c',
  green: '#107c10',
  greenLight: '#bad80a',*/

});
exports.DefaultPalette = DefaultPalette;
