// @skatteetaten/frontend-components/cjs/Card/Card.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Card = void 0;

var _classnames = _interopRequireDefault(require("classnames"));

var React = _interopRequireWildcard(require("react"));

var _Card = require("./Card.classNames");

var _Icon = require("../Icon");

var _Card2 = require("./Card.types");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/*
 * visibleName Card (Innholdskort)
 */
class Card extends React.PureComponent {
  constructor(props) {
    super(props);

    _defineProperty(this, "_toggleExpand", () => {
      this.setState({
        isExpandedState: !this.state.isExpandedState
      });
      this.props.onClick && this.props.onClick();
    });

    const {
      isExpanded
    } = props;
    this.state = {
      isExpandedState: isExpanded || false
    };
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state.isExpandedState !== prevState.isExpandedState) {
      this.props.onChange && this.props.onChange(this.state.isExpandedState);
    }
  }

  render() {
    const {
      isExpandedState
    } = this.state;

    const _this$props = this.props,
          {
      children,
      title,
      titleTagName,
      subtitle,
      expand,
      className,
      id,
      buttonType,
      ariaLabel,
      isExpanded,
      titlesize,
      color,
      border,
      marginbottom,
      margin,
      onChange,
      onClick
    } = _this$props,
          htmlAttributes = _objectWithoutProperties(_this$props, ["children", "title", "titleTagName", "subtitle", "expand", "className", "id", "buttonType", "ariaLabel", "isExpanded", "titlesize", "color", "border", "marginbottom", "margin", "onChange", "onClick"]);

    const TitleTag = titleTagName;
    const styles = (0, _Card.getClassNames)(this.props, this.state);
    const expandCard = /*#__PURE__*/React.createElement("button", {
      className: styles.header,
      onClick: expand ? this._toggleExpand : undefined,
      "aria-expanded": isExpandedState,
      type: buttonType
    }, /*#__PURE__*/React.createElement("div", {
      className: styles.titlecontainer
    }, /*#__PURE__*/React.createElement(TitleTag, {
      "aria-label": title,
      className: styles.titleExpand
    }, title), /*#__PURE__*/React.createElement("div", {
      className: styles.subtitle,
      "aria-label": subtitle
    }, subtitle)), /*#__PURE__*/React.createElement("div", {
      className: styles.expandIcon
    }, /*#__PURE__*/React.createElement(_Icon.Icon, {
      iconName: 'ChevronDown'
    })));
    const regularCard = /*#__PURE__*/React.createElement("div", {
      className: styles.header
    }, /*#__PURE__*/React.createElement("div", {
      className: styles.titlecontainer
    }, /*#__PURE__*/React.createElement(TitleTag, {
      className: styles.title,
      "aria-label": title
    }, title), /*#__PURE__*/React.createElement("div", {
      className: styles.subtitle,
      "aria-label": subtitle
    }, subtitle)));
    return /*#__PURE__*/React.createElement("div", _extends({
      id: id,
      className: (0, _classnames.default)(styles.root, className)
    }, htmlAttributes), expand ? expandCard : title || subtitle ? regularCard : null, isExpandedState && /*#__PURE__*/React.createElement("div", {
      className: styles.body
    }, children));
  }

}

exports.Card = Card;

_defineProperty(Card, "Color", _Card2.CardColor);

_defineProperty(Card, "Border", _Card2.CardBorder);

_defineProperty(Card, "defaultProps", {
  title: undefined,
  titleTagName: 'div',
  subtitle: undefined,
  titlesize: 'x-large',
  expand: false,
  isExpanded: true,
  color: _Card2.CardColor.BEIGE,
  marginbottom: '2px',
  margin: 'medium',
  buttonType: 'button'
});
